### ISSUE TITLE
Don't close floating window when rename or create files using floating `vim.ui.input` window

### ISSUE THREAD

**Issue by xvvhang** [nvim-tree/nvim-tree.lua#2607]
**Is this a question?**
N

**Can this functionality be implemented utilising API?**
I tried to solve the problem with API but I found it kind of difficult.

**Is your feature request related to a problem? Please describe.**
Yes. When using a floating window and create files or rename files (with dressing.nvim), nvim-tree's 'auto close when focus loss' feature make nvim-tree close and it's kind of annoying because I need to keep the window open. There's a video attached.

**Describe the solution you'd like**
Keep nvim-tree window open when rename or create files (with dressing.nvim)

**Describe alternatives you've considered**
N

**Additional context**
N

**Reproduce**

```lua
vim.g.loaded_netrw = 1
vim.g.loaded_netrwPlugin = 1

vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvt-min/site]]
local package_root = "/tmp/nvt-min/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup {
    {
      "wbthomason/packer.nvim",
      "nvim-tree/nvim-tree.lua",
      "nvim-tree/nvim-web-devicons",
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
      "stevearc/dressing.nvim"
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  }
end
if vim.fn.isdirectory(install_path) == 0 then
  print "Installing nvim-tree and dependencies."
  vim.fn.system { "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path }
end
load_plugins()
require("packer").sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua setup()]]
vim.opt.termguicolors = true
vim.opt.cursorline = true

-- MODIFY NVIM-TREE SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
_G.setup = function()
  require("nvim-tree").setup {
    view = {
      float = {
        enable = true
      }
    }
  }
  require("dressing").setup {}
end

-- UNCOMMENT this block for diagnostics issues, substituting pattern and cmd as appropriate.
-- Requires diagnostics.enable = true in setup.
--[[
vim.api.nvim_create_autocmd("FileType", {
  pattern = "lua",
  callback = function()
    vim.lsp.start { cmd = { "lua-language-server" } }
  end,
})
]]
```

https://github.com/nvim-tree/nvim-tree.lua/assets/46291413/c3d363b5-b7e7-4d52-ba4c-462ab71a89ef

**Comment by geril2207** [comment]
You can specify 
```lua
float = {
  enable = true,
  quit_on_focus_loss = false
}
```

**Comment by alex-courtis** [comment]
Nice one @geril2207 

Is this something we should attempt to "fix" or simply document this workaround?

**Comment by geril2207** [comment]
> Nice one @geril2207
> 
> Is this something we should attempt to "fix" or simply document this workaround?

Maybe, instead of opening files inside a floating window, we can open them as normal files, but I don't know if it's possible and should we.

**Comment by alex-courtis** [comment]
> Maybe, instead of opening files inside a floating window, we can open them as normal, but I don't know if it's possible and should we.

That could work. We have plumbing to control where buffers are opened: `open_file.eject` and `node.open.replace_tree_buffer`

`view._prevent_buffer_override` is solid, however the `vim.schedule` timing might be tricky.

**Comment by xvvhang** [comment]
With `quit_on_focus_loss` set to false, nvim-tree will not close automatically when I open files with `o` command. So it needs two behaviors actually when using floating window. One is for open files and one is for rename/create files. Let `quit_on_focus_loss` has no effect on rename/create actions would be nice. Adding some more configs to control behaviors would be better. 😄  @alex-courtis @geril2207

**Comment by geril2207** [comment]
Oh yeah, additionally you need to specify quit on open:
```lua
actions = {
  open_file = {
    quit_on_open = true,
  },
},

```

**Comment by xvvhang** [comment]
It works! Thank your for your help @geril2207. Maybe I could close this issue. And it would be better if you'd like to complete the docs to help other users. Thanks~ ❤️
